src/modules/wreck/wrexecd.c

```diff
@@ -1651,20 +1655,29 @@ static int l_push_environ (lua_State *L, int index)
     return (1);
 }
 
-static kvsdir_t *prog_ctx_kvsdir (struct prog_ctx *ctx)
+static flux_kvsdir_t *prog_ctx_kvsdir (struct prog_ctx *ctx)
 {
+    flux_t *h = prog_ctx_flux_handle (ctx);
     struct task_info *t;
 
     if (!ctx->in_task)
         return (ctx->kvs);
 
     t = prog_ctx_current_task (ctx);
     if (!t->kvs) {
-        if ( (kvs_get_dir (prog_ctx_flux_handle (ctx), &t->kvs,
-                           "%s.%d", ctx->kvspath, t->globalid) < 0)
-          && (errno != ENOENT))
-            wlog_err (ctx, "kvs_get_dir (%s.%d): %s",
-                      ctx->kvspath, t->globalid, flux_strerror (errno));
+        char *key = NULL;
+        flux_future_t *f = NULL;
+        const flux_kvsdir_t *dir;
+        if (asprintf (&key, "%s.%d", ctx->kvspath, t->globalid) < 0
+                    || !(f = flux_kvs_lookup (h, FLUX_KVS_READDIR, key))
+                    || flux_kvs_lookup_get_dir (f, &dir) < 0
+                    || !(t->kvs = flux_kvsdir_copy (dir))) {
+            if (errno != ENOENT)
```

Comment:
this chunk of code is repeated twice at least in this source file, maybe it could be its own static function? In general, the lookup->get_dir->copy_dir seems to be common enough idiom that I'm wondering why there isn't a function in the API for it, but it isn't quite common enough for me to care (especially if the flux_kvsdir_t is going to go away.)